### PR TITLE
[Fix]: Handle gracefully when device runs out of disk space

### PR DESCRIPTION
**Pull request by marcomaddy1982**
### Issues

See the [issue](https://wearezeta.atlassian.net/browse/SQCORE-429)

When there was a problem to load the database in case we run out of disk space (or for any other error), the database was dropped. This means that the user was losing all the previous conversation

NOTA BENE: The copies added in this PR are still not official. 

### Solutions

Now a screen similar to the `blacklisted` or `jailbroken` is shown but with a different message. Moreover the alert contains two buttons:

- Device Settings -> Which will show the application settings
- Delete Database -> When the user will choose it another confirmation alert will be show. If the user confirms the database will be removed and the login screen will be shown.

### Dependencies

data-model: https://github.com/wireapp/wire-ios-data-model/pull/1144
sync-engince:  https://github.com/wireapp/wire-ios-sync-engine/pull/1411

### CODE REVIEW

File: Wire-iOS/Resources/Base.lproj/Localizable.strings [hunk 1]
```diff
@@ -1626,6 +1626,10 @@
 "jailbrokendevice.alert.title" = "Jailbreak detected";
 "jailbrokendevice.alert.message" = "For security reasons, Wire can't be used on this device. Any existing Wire data has been erased.";
 
+// Database Loading Failure alert
+"databaseloadingfailure.alert.title" = "Database loading failure";
+"databaseloadingfailure.alert.message" = "The database could be loaded.";
```
Comment (typfel) [inline Wire-iOS/Resources/Base.lproj/Localizable.strings:1631]:
```suggestion
"databaseloadingfailure.alert.message" = "The database could not be loaded.";
```

Comment (typfel) [reply Wire-iOS/Resources/Base.lproj/Localizable.strings:1631]: This copy is pending a review?

**Review by typfel — CHANGES_REQUESTED** [review]
Look like some linters warnings was introduced in AppStateCalculator

File: Wire-iOS/Resources/Base.lproj/Localizable.strings (hunk 1 above)
Comment (marcomaddy1982) [reply Wire-iOS/Resources/Base.lproj/Localizable.strings:1631]: Yes, this is just temp copy. Better if I mention on the ticket description.

**Comment by KaterinaWire** [comment]
@marcomaddy1982 Everything is fine for me, I don't have comments. Could you notify when the copies will be approved by the design team?

**Review by KaterinaWire — APPROVED** [review]

**Comment by marcomaddy1982** [comment]
> Look like some linters warnings was introduced in AppStateCalculator

Solved

File: Wire-iOS/Sources/UserInterface/BlockerViewController.swift [hunk 2]
```diff
@@ -47,17 +51,77 @@ final class BlockerViewController: LaunchImageViewController {
...
 
     func showJailbrokenMessage() {
-        presentAlertWithOKButton(title: "jailbrokendevice.alert.title".localized, message: "jailbrokendevice.alert.message".localized)
+        presentAlertWithOKButton(title: "jailbrokendevice.alert.title".localized,
+                                 message: "jailbrokendevice.alert.message".localized)
     }
 
+    func showDatabaseFailureMessage() {
+        let databaseFailureAlert = UIAlertController(
+            title: "databaseloadingfailure.alert.title".localized,
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/BlockerViewController.swift:73]: What about using the type safe localization John introduced in https://github.com/wireapp/wire-ios/pull/4911

Comment (marcomaddy1982) [reply Wire-iOS/Sources/UserInterface/BlockerViewController.swift]: Done

**Review by typfel — APPROVED** [review]

**Review by typfel — APPROVED** [review]

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**